pkg/controllermanager/controller/seed/seed_extension_check_control.go

```diff
@@ -147,47 +159,62 @@ func (r *extensionCheckReconciler) reconcile(ctx context.Context, seed *gardenco
-	if err != nil {
-		return err
-	}
-
-	if condition := helper.GetCondition(seed.Status.Conditions, gardencorev1beta1.SeedExtensionsReady); condition != nil {
-		bldr.WithOldCondition(*condition)
-	}
+	condition := helper.GetOrInitCondition(seed.Status.Conditions, gardencorev1beta1.SeedExtensionsReady)
+	thresholdMappings := r.conditionThresholdsToProgressingMapping()
 
 	switch {
 	case len(notValid) != 0:
-		bldr.
-			WithStatus(gardencorev1beta1.ConditionFalse).
-			WithReason("NotAllExtensionsValid").
-			WithMessage(fmt.Sprintf("Some extensions are not valid: %+v", notValid))
-
+		condition = r.failedCondition(thresholdMappings, condition, "NotAllExtensionsValid", fmt.Sprintf("Some extensions are not valid: %+v", notValid))
 	case len(notInstalled) != 0:
-		bldr.
-			WithStatus(gardencorev1beta1.ConditionFalse).
-			WithReason("NotAllExtensionsInstalled").
-			WithMessage(fmt.Sprintf("Some extensions are not installed: %+v", notInstalled))
-
+		condition = r.failedCondition(thresholdMappings, condition, "NotAllExtensionsInstalled", fmt.Sprintf("Some extensions are not installed: %+v", notInstalled))
 	case len(notHealthy) != 0:
-		bldr.
-			WithStatus(gardencorev1beta1.ConditionFalse).
-			WithReason("NotAllExtensionsHealthy").
-			WithMessage(fmt.Sprintf("Some extensions are not healthy: %+v", notHealthy))
-
+		condition = r.failedCondition(thresholdMappings, condition, "NotAllExtensionsHealthy", fmt.Sprintf("Some extensions are not healthy: %+v", notHealthy))
 	default:
-		bldr.
-			WithStatus(gardencorev1beta1.ConditionTrue).
-			WithReason("AllExtensionsReady").
-			WithMessage("All extensions installed into the seed cluster are ready and healthy.")
+		condition = helper.UpdatedCondition(condition, gardencorev1beta1.ConditionTrue, "AllExtensionsReady", "All extensions installed into the seed cluster are ready and healthy.")
 	}
 
 	// patch ExtensionsReady condition
 	patch := client.StrategicMergeFrom(seed.DeepCopy())
-	newCondition, needsUpdate := bldr.WithNowFunc(r.nowFunc).Build()
-	if !needsUpdate {
+	newConditions := helper.MergeConditions(seed.Status.Conditions, condition)
+	if !helper.ConditionsNeedUpdate(seed.Status.Conditions, newConditions) {
 		return nil
 	}
-	seed.Status.Conditions = helper.MergeConditions(seed.Status.Conditions, newCondition)
+	seed.Status.Conditions = newConditions
 	return r.gardenClient.Status().Patch(ctx, seed, patch)
 }
+
+func (r *extensionCheckReconciler) conditionThresholdsToProgressingMapping() map[gardencorev1beta1.ConditionType]time.Duration {
+	out := make(map[gardencorev1beta1.ConditionType]time.Duration)
+	for _, threshold := range r.config.ConditionThresholds {
+		out[gardencorev1beta1.ConditionType(threshold.Type)] = threshold.Duration.Duration
+	}
+	return out
+}
+
+// FailedCondition returns a progressing or false condition depending on the progressing threshold.
+func (r *extensionCheckReconciler) failedCondition(
+	conditionThresholds map[gardencorev1beta1.ConditionType]time.Duration,
```

Comment:
Can you rather pass only the threshold duration for this specific condition instead of the full map? Then you can replace statements like this

```golang
		if _, ok := conditionThresholds[condition.Type]; !ok {
```

with

```golang
		if conditionThreshold == 0 {
```